python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -633,6 +646,98 @@ def is_successful(self) -> bool:
 
         return self.process.wait() == 0
 
+    def _stream_asset_events(
+        self,
+    ) -> Iterator[DbtDagsterEventType]:
+        """Stream the dbt CLI events and convert them to Dagster events."""
+        for event in self.stream_raw_events():
+            yield from event.to_default_asset_events(
+                manifest=self.manifest,
+                dagster_dbt_translator=self.dagster_dbt_translator,
+                context=self.context,
+                target_path=self.target_path,
+            )
+
+    def _get_dbt_resource_props_from_event(self, event: DbtDagsterEventType) -> Dict[str, Any]:
+        unique_id = cast(TextMetadataValue, event.metadata["unique_id"]).text
+        return check.not_none(self.manifest["nodes"].get(unique_id))
+
+    def _attach_post_materialization_metadata(
+        self,
+        event: DbtDagsterEventType,
+    ) -> DbtDagsterEventType:
+        """Threaded task which runs any postprocessing steps on the given event before it's
+        emitted to user code.
+
+        This is used to, for example, query the row count of a table after it has been
+        materialized by dbt.
+        """
+        adapter = check.not_none(self.adapter)
+
+        dbt_resource_props = self._get_dbt_resource_props_from_event(event)
+        is_view = dbt_resource_props["config"]["materialized"] == "view"
+
+        # Avoid counting rows for views, since they may include complex SQL queries
+        # that are costly to execute. We can revisit this in the future if there is
+        # a demand for it.
+        if is_view:
+            return event
+
+        # If the adapter is DuckDB, we need to wait for the dbt CLI process to complete
+        # so that the DuckDB lock is released
+        try:
+            from dbt.adapters.duckdb import DuckDBAdapter
+
+            if isinstance(adapter, DuckDBAdapter):
+                self._dbt_run_thread.result()
+        except ImportError:
+            pass
+
+        unique_id = dbt_resource_props["unique_id"]
+        table_str = f"{dbt_resource_props['database']}.{dbt_resource_props['schema']}.{dbt_resource_props['name']}"
+
+        adapter = check.not_none(self.adapter)
+
+        with adapter.connection_named(f"row_count_{unique_id}"):
+            query_result = adapter.execute(
+                f"""
+                    SELECT
+                    count(*)
+                    FROM
+                    {table_str}
+                """,
+                fetch=True,
+            )
+        row_count = query_result[1][0][0]
```

Comment:
nit: can we alias `count(*)` so that we don't have to do this array indexing? Or could we use result de-structuring instead?